src/index.test.ts

```diff
@@ -67,5 +68,119 @@ describe('Geckoboard', () => {
     expect(async () => await gb.ping()).rejects.toThrow(
       new Error('Something went wrong with the request'),
     );
+    expect(() => mockAgent.assertNoPendingInterceptors()).not.toThrow();
+  });
+
+  it('can create a dataset', async () => {
+    const mockAgent = new MockAgent();
+    setGlobalDispatcher(mockAgent);
+
+    const mockPool = mockAgent.get('https://api.geckoboard.com');
+    mockPool
+      .intercept({
+        method: 'PUT',
+        path: '/datasets/steps.by.day',
+        headers: {
+          Authorization: `Basic ${btoa('API_KEY:')}`,
+          'User-Agent': 'Geckoboard Node Client 2.0.0',
+          'Content-type': 'application/json',
+        },
+        body: JSON.stringify({
+          fields: {
+            steps: {
+              type: 'number',
+              name: 'Steps',
+              optional: false,
+            },
+            timestamp: {
+              type: 'datetime',
+              name: 'Date',
+            },
+          },
+          unique_by: ['timestamp'],
+        }),
+      })
+      .reply(
+        200,
+        JSON.stringify({
+          id: 'steps.by.day',
+          fields: {
+            steps: {
+              type: 'number',
+              name: 'Steps',
+              optional: false,
+            },
+            timestamp: {
+              type: 'datetime',
+              name: 'Date',
+            },
+          },
+          unique_by: ['timestamp'],
+        }),
+      );
+
+    const gb = new Geckoboard('API_KEY');
+    const dataset = gb.defineDataset({
+      id: 'steps.by.day',
+      fields: {
+        steps: {
+          type: 'number',
+          name: 'Steps',
+          optional: false,
+        },
+        timestamp: {
+          type: 'datetime',
+          name: 'Date',
+        },
+      },
+      uniqueBy: ['timestamp'],
```

Comment:
are you allowed to be uniqueBy on an optional field?

Reply:
Good question, I'll check.

Reply:
You get this error back from the API : ''You cannot use "timestamp" in "unique_by" as this field is optional'. To be clear though, the default is the field is mandatory and you have to include `optional: true` in order to make a field optional. Using `optional: false` here is probably a bad example.